File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> p_q = new PriorityQueue<>(Comparator.reverseOrder());
        
        for(int item : nums){
            p_q.add(item);
        }
        
        while(k-- != 1){
            p_q.poll();
        }
        
        return p_q.peek();
    }
}

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(std::vector<int>& nums, int k) {
        std::priority_queue<int, std::deque<int>, std::greater<int>> p_q;

        for (int i : nums) {
            p_q.push(i);

            if (p_q.size() == k + 1) {
                p_q.pop();
            }
        }

        return p_q.top();
    }
};